### PR TITLE
Read package names from a JSON/YAML file

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

**Comment by fridex** [comment]
```
python3 ./app.py --package-names-file adviser-result.json --package-names-file-jsonpath 'result.report._ERROR_DETAILS.unresolved'
```

The `app.py` also accepts environment variables `THOTH_PACKAGE_RELEASES_PACKAGE_NAMES_FILE` and `THOTH_PACKAGE_RELEASES_PACKAGE_NAMES_FILE_JSONPATH`.

**Comment by fridex** [comment]
I had to remove pre-commit from the stack. It causes troubles when locking as it requires PyYAML>=5.1, where other dependencies depend on a compliant version to 3.11.

I don't think it should go to the application requirements anyway:

1. it's not used to actually test the application. It is a tool to support git management and operations (the same reason why git would not be present in the requirements if git would be in Python).
2. one can operate on git outside of virtual environment, in that case no pre-commit would be found unless there is installed one in the environment itself
3. using pre-commit is optional on user side, checks should be done in CI anyway to guarantee users contribute to project correctly

File: app.py [hunk 1]
```diff
@@ -265,9 +281,43 @@ def cli(
...
+        package_names = []
+        for item in parse(package_names_file_jsonpath).find(content):
+            for entry in item.value:
+                if not isinstance(entry, str):
+                    raise ValueError(f"Found item in the file is not a string describing package name: {item!r}")
+                package_names.append(entry)
+
+        if not package_names:
+            _LOGGER.warning("No packages to be checked for new releases")
+            sys.exit(2)
```
Comment (fridex) [inline app.py]: @pacospace Would this fit into the argo workflow?

Comment (pacospace) [reply app.py]: It should fit!! Thanks @fridex !!

**Pull request merged**